modules/lib/xml.nix

```diff
@@ -0,0 +1,36 @@
+{ libxslt, libxml2, runCommand }:
+
+rec {
+  genXMLFile = input:
+    runCommand "generated-xml" {
+      input = builtins.toXML input;
+
+      prefix = ''
+        <?xml version="1.0" encoding="utf-8"?>
+      '' + (if input ? doctype then
```

Comment:
What about `lib.optionalString` instead?

Reply:
I wasn't aware that existed.